### PR TITLE
Implement decimal comparisons as vector functions

### PR DESCRIPTION
**Pull request by majetideepak**
Decimal Simple Functions require separate signatures/registration for
ShortDecimal and LongDecimal types.
We want to view Decimal Types as a single entity/type. Simple Functions break this view.
To achieve the single entity view, all Decimal functions must be implemented as Vector Functions.
The Decimal comparison functions are currently implemented as Simple Functions.
The scope of this PR is to implement them as Vector Functions.

### CODE REVIEW

**Review by mbasmanova — COMMENTED** [review]
@Yuhta Jimmy, would you help review this PR?

**Comment by karteekmurthys** [comment]
@majetideepak Since comparisons are re-written as vector functions, should we handle rescaling of inputs? to keep it more generic.

**Comment by majetideepak** [comment]
@karteekmurthys The current implementation extends the SimdComparison vector functions used by other scalar types.
Rescaling will not easily fit here. I also don't think additional complexity is required here.

**Review by karteekmurthys — APPROVED** [review]
LGTM! Once we migrate to using BIGINT and HUGEINT for decimals we should be able leverage SIMD. Thanks @majetideepak.

**Comment by majetideepak** [comment]
@Yuhta I rebase with the main branch. Can you take a look? Thanks.

File: velox/functions/prestosql/Comparisons.cpp
Comment (Yuhta) [inline velox/functions/prestosql/Comparisons.cpp:100]: Extract this condition out so we can reuse it in different specialization of the template

File: velox/functions/prestosql/Comparisons.cpp
Comment (karteekmurthys) [reply velox/functions/prestosql/Comparisons.cpp:119]: These changes will be overwritten with Bigint and huge INR representations after this Pr is merged https://github.com/facebookincubator/velox/pull/4434

Comment (Yuhta) [reply velox/functions/prestosql/Comparisons.cpp:119]: We probably should merge https://github.com/facebookincubator/velox/pull/4434 first then?

Comment (majetideepak) [reply velox/functions/prestosql/Comparisons.cpp:119]: We need this PR to be merged before #4434 since there is no provision for decimal scalar functions there.
As @karteekmurthys said, we will be replacing UnscaledShortDecimal with int64_t and UnscaledLongDecimal with int128_t. So we can keep this as is for now.

File: velox/functions/prestosql/Comparisons.cpp
Comment (majetideepak) [reply velox/functions/prestosql/Comparisons.cpp:100]: I will be reverting this change in #4434 as well. Do you still want me to extract this condition?

Comment (Yuhta) [reply velox/functions/prestosql/Comparisons.cpp:100]: That's good then

**Comment by majetideepak** [comment]
@Yuhta can you please help merge this by EOD today? Thanks!

**Comment by majetideepak** [comment]
@Yuhta @mbasmanova Can you please help merge this? Let me know if I can help with something. Thanks.

**Pull request closed without merging**
